sycl/source/device.cpp

```diff
@@ -208,5 +208,78 @@ pi_native_handle device::getNative() const { return impl->getNative(); }
 
 bool device::has(aspect Aspect) const { return impl->has(Aspect); }
 
+void device::ext_oneapi_enable_peer_access(const device &peer) {
+  const RT::PiDevice Device = impl->getHandleRef();
+  const RT::PiDevice Peer = peer.impl->getHandleRef();
+  if (Device != Peer) {
+    auto Plugin = impl->getPlugin();
+    pi_result result =
+        Plugin->call_nocheck<detail::PiApiKind::piextEnablePeerAccess>(Device,
+                                                                       Peer);
+    if (result != PI_SUCCESS) {
+      char *message = nullptr;
+      Plugin->call<detail::PiApiKind::piPluginGetLastError>(&message);
+      throw sycl::exception(make_error_code(errc::runtime), message);
+    }
+  }
+}
+
+void device::ext_oneapi_disable_peer_access(const device &peer) {
+  const RT::PiDevice Device = impl->getHandleRef();
+  const RT::PiDevice Peer = peer.impl->getHandleRef();
+  if (Device != Peer) {
+    auto Plugin = impl->getPlugin();
+    pi_result result =
+        Plugin->call_nocheck<detail::PiApiKind::piextDisablePeerAccess>(Device,
+                                                                        Peer);
+    if (result != PI_SUCCESS) {
+      char *message = nullptr;
+      Plugin->call<detail::PiApiKind::piPluginGetLastError>(&message);
+      throw sycl::exception(make_error_code(errc::runtime), message);
+    }
+  }
+}
+
+bool device::ext_oneapi_can_access_peer(const device &peer,
+                                        ext::oneapi::peer_access attr) {
+  const RT::PiDevice Device = impl->getHandleRef();
+  const RT::PiDevice Peer = peer.impl->getHandleRef();
+
+  if (Device == Peer) {
+    return true;
+  }
+
+  if (peer.get_backend() != backend::ext_oneapi_cuda) {
+    return false;
+  }
```

Comment:
I would prefer that we instead make some basic implementation of the `piextPeerAccessGetInfo` for all the backends and instead let them decide when they are ready to support it, rather than making the restriction at runtime-level. Is there a reason not to do that?